# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/DOMIntersectionObserver.h
# Commit: 20a5c1f3b35e
# Full Hash: 20a5c1f3b35e80849e0b98ffbe3f9488d703717b
# Author: Frederic Wang <fred.wang@free.fr>
# Date: 2023-11-22 21:50:37
# Regressor Bug: 1807253
# File Overlap Count: 1
# Description:
#   Bug 1807253 - unreliable timing for content-visibility:auto, r=emilio
#   
#   Rewrite implementation of content-visibility: auto as defined in
#   https://github.com/w3c/csswg-drafts/issues/8542
#   
# ==============================================================================

diff -r b90083f581e1 -r 20a5c1f3b35e dom/base/DOMIntersectionObserver.h
--- a/dom/base/DOMIntersectionObserver.h	Mon Nov 20 11:33:16 2023 +0000
+++ b/dom/base/DOMIntersectionObserver.h	Wed Nov 22 13:00:36 2023 +0000
@@ -152,10 +152,10 @@
       const Document& aDocument, const nsINode* aRoot,
       const StyleRect<LengthPercentage>* aRootMargin);
 
-  enum class IsContentVisibilityObserver : bool { No, Yes };
+  enum class IsForProximityToViewport : bool { No, Yes };
   static IntersectionOutput Intersect(
       const IntersectionInput&, const Element&,
-      IsContentVisibilityObserver = IsContentVisibilityObserver::No);
+      IsForProximityToViewport = IsForProximityToViewport::No);
   // Intersects with a given rect, already relative to the root frame.
   static IntersectionOutput Intersect(const IntersectionInput&, const nsRect&);
 
@@ -165,9 +165,6 @@
   static already_AddRefed<DOMIntersectionObserver> CreateLazyLoadObserver(
       Document&);
 
-  static already_AddRefed<DOMIntersectionObserver>
-  CreateContentVisibilityObserver(Document&);
-
   static Maybe<nsRect> EdgeInclusiveIntersection(const nsRect& aRect,
                                                  const nsRect& aOtherRect);
 